addons/website/models/website_page.py

```diff
@@ -229,6 +238,94 @@ def _set_cache_response(self, cache_key, response):
         """ Put in cache the given response. """
         self.pool._Registry__cache[('website.page', _cached_response, self.id, cache_key)] = response
 
+    @api.model
+    def _search_get_detail(self, website, order, options):
+        with_description = options['displayDescription']
+        requires_sudo = False
+        domain = [website.website_domain()]
+        if not self.env.user.has_group('website.group_website_designer'):
+            domain.append([('website_published', '=', True)])
+            requires_sudo = True
```

Comment:
At that point, `requires_sudo` should be true for all users, why special-case it for `group_website_designer`? Best case, they take a small perf hit due to ACL checks, worst case, they don't see the same results as everybody else, no?

Reply:
and the use of `requires_sudo` by itself deserves a comment, as it's unclear why `website.page` is getting a special treatment for that